src/builders/indexable-post-builder.php

```diff
@@ -415,4 +407,19 @@ protected function get_number_of_pages_for_post( $post ) {
 	protected function should_exclude_post( $post ) {
 		return $this->post_type_helper->is_excluded( $post->post_type );
 	}
+
+	/**
+	 * Transforms an empty string into null. Leaves non-empty strings intact.
+	 *
+	 * @param string $string The string.
+	 *
+	 * @return string|null The input string or null.
+	 */
+	protected function empty_string_to_null( $string ) {
+		if ( empty( $string ) ) {
```

Comment:
Eh... you sure `empty()` is what you want here ? Would string `'0'` always be invalid and should it be treated as `null` ?

There is also no _guarantee_ that the input is actually a string or that the output will be `string|null`, even though it is "most likely", but only as long as there is still a tight coupling with `WPSEO_Meta` and considering there is now a helper in between and that helper is injected in the constructor, you don't actually have that guarantee.

From `WPSEO_Meta::get_value()`: 
```php
 	 * @return string All 'normal' values returned from get_post_meta() are strings.
	 *                Objects and arrays are possible, but not used by this plugin
	 *                and therefore discarded (except when the special 'serialized' field def
	 *                value is set to true - only used by add-on plugins for now).
	 *                Will return the default value if no value was found.
	 *                Will return empty string if no default was found (not one of our keys) or
	 *                if the post does not exist.
	 */
	public static function get_value( $key, $postid = 0 ) {...}
```

Lastly: should this method be `protected` or `private` ?

Reply:
Great point! 
Ugh `var_dump( empty( '0' ) ); // true` I'm not sure how I feel about that 😅
I'll change that and add will find a better way to deal with different types 👍